### PR TITLE
FEATURE: ONE-2217 Allow running in nodejs

### PR DESCRIPTION
**Pull request by pbenes**

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul/323/ : FAILURE in 1m 53s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/88/ : FAILURE in 1m 08s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul/324/ : FAILURE in 3m 16s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/89/ : FAILURE in 2m 24s

**Comment by vaclavbohac** [comment]
@pbenes needs to be rebased

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul/333/ : FAILURE in 1m 00s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/98/ : FAILURE in 47s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul/334/ : FAILURE in 1m 20s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/99/ : FAILURE in 52s

File: src/project.js [hunk 1]
```diff
@@ -152,3 +153,83 @@ export function setTimezone(projectId, timezone) {
...
+    return projectState === 'ENABLED' ||
+        projectState === 'DELETED';
+};
+
+const waitUntilProjectEnabled = (response, options = {}) => {
+    const { pollStep = 5000 } = options;
+    const projectUri = response.uri;
+
+    return new Promise((resolve, reject) => {
+        AsyncPolling(function(end) { // eslint-disable-line func-names
```
Comment (vaclavbohac) [inline src/project.js]: I know it's still far from finished, but that polling code seems too complicated.

Assuming using bluebird, we could do something like:
```js
const Promise = require('bluebird');

function poll() {
    return getProject().then((project) => {
        if (isProjectCreated(project)) {
            return Promise.resolve();
        }

        return Promise.delay(5000).then(poll);
    });
}
```

File: src/project.js [hunk 2]
```diff
@@ -152,3 +153,83 @@ export function setTimezone(projectId, timezone) {
...
+        title = 'Project',
+        summary = 'Project',
+        projectTemplate = '/projectTemplates/GoodSalesDemo/2',
+        driver = 'Pg',
+        environment = 'TESTING',
+        guidedNavigation = 1,
+        authorizationToken
+    } = options;
+
+    return new Promise((resolve, reject) => {
```
Comment (vaclavbohac) [inline src/project.js]: `post` itself returns a promise, so there's no need to create a new promise

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul/335/ : FAILURE in 1m 58s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/100/ : FAILURE in 48s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul/336/ : FAILURE in 1m 13s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/101/ : FAILURE in 53s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul/337/ : SUCCESS in 1m 52s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/102/ : SUCCESS in 1m 10s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul/338/ : SUCCESS in 1m 55s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/103/ : SUCCESS in 1m 43s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul/340/ : SUCCESS in 46s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/105/ : SUCCESS in 41s

File: src/project.js [hunk 3]
```diff
@@ -1,7 +1,9 @@
 // Copyright (C) 2007-2014, GoodData(R) Corporation. All rights reserved.
-import { ajax, get, put, parseJSON } from './xhr';
+import { post, ajax, get, del, put, parseJSON } from './xhr';
 import { getIn } from './util';
 
+const Promise = require('bluebird');
```
Comment (vaclavbohac) [inline src/project.js]: import?

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul/341/ : SUCCESS in 45s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/106/ : SUCCESS in 40s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul/342/ : SUCCESS in 1m 51s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/107/ : SUCCESS in 30s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul/344/ : SUCCESS in 2m 27s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/109/ : SUCCESS in 1m 40s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul/345/ : SUCCESS in 2m 33s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/110/ : SUCCESS in 44s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul/347/ : SUCCESS in 44s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/112/ : SUCCESS in 33s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul/348/ : SUCCESS in 49s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/113/ : SUCCESS in 44s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul/350/ : SUCCESS in 1m 53s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/115/ : SUCCESS in 42s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul/351/ : SUCCESS in 58s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/116/ : SUCCESS in 41s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul/352/ : SUCCESS in 58s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/117/ : SUCCESS in 44s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul/353/ : SUCCESS in 1m 46s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/118/ : SUCCESS in 35s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul/355/ : SUCCESS in 55s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/120/ : SUCCESS in 47s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul/356/ : SUCCESS in 2m 36s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/121/ : SUCCESS in 2m 34s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul/357/ : SUCCESS in 1m 53s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/122/ : SUCCESS in 54s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul/358/ : SUCCESS in 4m 25s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/123/ : SUCCESS in 2m 01s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul/371/ : FAILURE in 1m 27s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/136/ : FAILURE in 46s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul/372/ : SUCCESS in 2m 09s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/137/ : SUCCESS in 2m 06s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://ci.intgdc.com/job/gooddata-js-merge-compliance-check/11/ : SUCCESS in 12s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul/373/ : SUCCESS in 1m 46s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/138/ : SUCCESS in 1m 04s

**Pull request merged**